Adding Dynamic Form Field Configuration Example

ChrisDecombe opened the pull request:
After discussion we Josh we decided to use the existing Github repository to prepare and publish an example of a Dynamic Form Fields configuration.

vpstoynova commented on Dynamic Form Fields Configuration Example/README.md:
```diff
@@ -0,0 +1,32 @@
+
+## How to use this example
+
+Before using this example file, you should check and update his content:
```
```suggestion
Before using this example file, you should check and update the following content:
```

vpstoynova commented on Dynamic Form Fields Configuration Example/README.md:
```diff
@@ -0,0 +1,32 @@
+
+A Dynamic Form Field configuration is used to display custom fields inside 
+the following resources in SystemLink Enterprise: Work order, Test plan, 
+Product, Asset and System. 
```
```suggestion
Product, Asset, and System. 
```

ChrisDecombe commented:
@adamarnesen I need a reviewer for this changes in SLE example repo.
I have added a new section with an example of the DFF configuration.

Could you check my pull request?
Josh has already checked it.

This DFF Configuration example would look this this:

![image](https://github.com/user-attachments/assets/ceecc728-43d8-42a1-93af-a4750505f339)

prestwick reviewed (commented):
It may be worthwhile is have several distinct examples rather than this one big example.

ChrisDecombe commented:
> It may be worthwhile is have several distinct examples rather than this one big example.

I have just removed the second group to keep it simple.
I have kept one example of each field type in the single group...
Would it be ok for you like this?

Josh wanted to have several groups that's why I have added a second group BUT I agree we should have an simple example.

ChrisDecombe commented:
@prestwick  I have cleaned and updated the DFF configuration example (with the new DFF field type Date) and add a link to the NI Documentation for DFF in the README.
As next I may add one configuration for each type of field with more information what can be set specifically for that type of DFF.

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+
+## Dynamic Form Fields Configuration
+
+A Dynamic Form Field configuration is used to display custom fields inside 
```
make plural

ChrisDecombe replied:
changed

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+
+The [DynamicFormFieldConfigurationExample.json](DynamicFormFieldConfigurationExample.json) 
+provides an example of a Dynamic Form Fields configuration with 1 view,
+1 group, and one example from each available type of fields:
```
field (singular)

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+
+The [DynamicFormFieldConfigurationExample.json](DynamicFormFieldConfigurationExample.json) 
+provides an example of a Dynamic Form Fields configuration with 1 view,
+1 group, and one example from each available type of fields:
```
suggest changing 
from each available type
to
for several types

This way, when new types are added in SLE this sentence is immediately outdates.

ChrisDecombe replied:
changed

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+  - Date Field
+
+This Dynamic Form Field configuration will appear only after having added a Custom Property 
+named "example" with the value set to "1" (displayRule).
```
how about:

This example Dynamic Form Field configuration has a display rule that looks for a custom property named "example" with a value of "1". Thus it will only appear after adding a Custom Property named "example" with a value of "1" to a specific SLE page.

ChrisDecombe replied:
updated

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+
+Before using this example file, you should read the [Documentation](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/adding-custom-fields-to-the-ui.html) and:
+
+  - update all workspace IDs to the correct one
```
update all workspace IDs to match the desired workspace(s)

I am assuming the workspace property can take multiple workspaces. If not, remove the "(s)"

ChrisDecombe replied:
I leave workspace without s because we recommend to use only one workspace inside a configuration.

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+Before using this example file, you should read the [Documentation](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/adding-custom-fields-to-the-ui.html) and:
+
+  - update all workspace IDs to the correct one
+  - check and update the resource type to define the resource in which the
```
resource > resource(s)

ChrisDecombe replied:
I have added the name of the parameter inside the json file.

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+
+  - update all workspace IDs to the correct one
+  - check and update the resource type to define the resource in which the
+    Dynamic Form Fields configuration will appear,
```
will > should

ChrisDecombe replied:
changed

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+
+## How to use this example
+
+Before using this example file, you should read the [Documentation](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/adding-custom-fields-to-the-ui.html) and:
```
How about:

Refer to the SystemLink Enterprise User Manual topic [Adding Custom Input Fields to the User Interface](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/adding-custom-fields-to-the-ui.html) for information about configuring dynamic form fields and making them available in SystemLink Enterprise. 

Make the following updates to the json example before trying it in SystemLink Enterprise:

ChrisDecombe replied:
changed

debryant commented:
> > It may be worthwhile is have several distinct examples rather than this one big example.
> 
> I have just removed the second group to keep it simple. I have kept one example of each field type in the single group... Would it be ok for you like this?
> 
> Josh wanted to have several groups that's why I have added a second group BUT I agree we should have an simple example.

What about adding documentation to the readme that describes how to add additional groups?  
Possibly at the end of the "Dynamic Form Fields Configuration" section. Something like (assuming this is correct):

This example can be extended to include additional groups by adding more groups to the Group section and referencing them in the configurations > views > groups property.

debryant commented on Dynamic Form Fields Configuration Examples/README.md:
```diff
@@ -0,0 +1,36 @@
+
+  - update all workspace IDs to the correct one
+  - check and update the resource type to define the resource in which the
+    Dynamic Form Fields configuration will appear,
```
Suggest listing the text for the options since this is an enum type of field. The resourceType property values options include: workorder, testplan, ....

ChrisDecombe replied:
I have a added a link to the NI Documentation in which the user can find the list of ResourceTypes

ChrisDecombe commented:
@prestwick, @joshuaprewitt 
I need a reviewer with write access to that repository to be able to merge this PR to add the DFF Configuration example.
Deborah has already reviewed it.
